lib/internal/streams/readable.js

```diff
@@ -234,6 +236,10 @@ Readable.prototype[EE.captureRejectionSymbol] = function(err) {
   this.destroy(err);
 };
 
+Readable.prototype[SymbolDispose] = function() {
+  this.destroy(new AbortError());
```

Comment:
```suggestion
Readable.prototype[SymbolAsyncDispose] = function() {
  this.destroy(new AbortError());
  return stream.promises.finished(this);
```
```

Reply:
Interesting, and this always works?

Reply:
It should, yes. Otherwise you have the risk of unhandler errors

Reply:
your code always rejects with the `AbortError` we destroy with

Reply:
modify it so that it doesn't reject abort errors I guess?

Reply:
Yeah I pushed some fix